orttraining/orttraining/core/graph/training_op_defs.cc

```diff
@@ -3291,7 +3291,7 @@ Return true if all elements are true and false otherwise.
 
         // Add support for 1D input X, in which case num_channels should default to 1.
         auto& input_shape = getInputShape(ctx, 0);
-        if (input_shape.dim_size() <= 1) {
+        if (input_shape.dim_size() == 1) {
           num_channels.set_dim_value(1);
         } else {
           unifyInputDim(ctx, 0, 1, num_channels);
```

Comment:
Does [defs in onnx](https://github.com/onnx/onnx/blob/8c526e3325591c229fd0e5fca4e8112766834bfe/onnx/defs/nn/defs.cc#L1741-L1746) also correctly handle such situation?

```c++
          if (hasInputShape(ctx, 0)) {
            if (getInputShape(ctx, 0).dim_size() > 1)
              unifyInputDim(ctx, 0, 1, num_channels);
            else
              unifyDim(num_channels, 1);
          }
```

Reply:
Yes, I think that should also work. Let me update this pull request so that the check is similar to BatchNormalization op.